### DISCUSSION TITLE
Is anyone having issues with expand since v0.8+?

### DISCUSSION THREAD

**Discussion by charbs-io** [pocketbase/pocketbase#934] [Q&A]
My responses are now returning with no `expand`...

**Comment by ganigeorgiev** [comment]
Could you check in the browser's "Inspect tool > Network" tab what is the response of the expand request?

It is also possible that you may need to adjust the "View API rule" in the expanded collection.

Also note, that in v0.8.0 the expanded relation are now under the `expand` prop (without the `@` prefix).

**Reply by charbs-io** [reply]
The `expand` object was being omitted completely. 

After some debugging, I have found that this rule on View is problematic: `@request.auth.id != ""`... If I leave the View rule blank, it works (i.e the expand object is returned and it's populated with the relation i'm looking for.

This used to be `@request.user.id != ""` in v0.7 - I know the migration updated it to `auth` - any idea if something broke during my migration or something?

**Reply by ganigeorgiev** [reply]
`@request.auth.id` is the correct one. Are you sure that you are authenticated?

In v0.8 the `Authorization` header no longer need a schema and it is in the format: `Authorization: YOUR_TOKEN`. If you are using the SDK the header is added automatically for you if  there is an auth model and token in the AuthStore.

**Reply by ganigeorgiev** [reply]
@cayoub88 It is also possible that you are hitting a small existing issue in the current release candidate - everytime when you restart the PocketBase server it will invalidate the users tokens and you'll have to relogin (see https://github.com/pocketbase/pocketbase/discussions/911). It is already fixed in the `rc` branch and will be available with the final v0.8.0 release, but for now you can just go to your "Admin UI > Settings" and do some small change to trigger Settings update.

**Reply by charbs-io** [reply]
Definitely authenticated. 

Using the SDK. I think this is my own mistake, somewhere in code or in the rules. However I noticed one thing.

Prior to migrating, I was using the following rule to check whether I can return certain items to the user:

`@request.user.id!="" && @request.user.profile.family.id = family.id`

I had added a relation in the now deprecated `profile` called family. This is how I was checking whether to return items for a user that belonged to a `family`, eventhough they didn't "own" the record on the table. 

Now, it seems as though I have no access to `family` through `@request.auth`, eventhough i made sure to add the `family` relation to the new `user` table. 

edit: trying to do something like this, but not working. No idea how to fix it... `@request.auth.id != '' && family.id = @collection.users.family.id`

Really appreciate your help so far.

**Reply by ganigeorgiev** [reply]
The `upgrade` command should have migrated your `family` relation and the above rule should look something like:

```
@request.auth.id!="" && @request.auth.family.id = family.id
```

> Now, it seems as though I have no access to family through @request.auth, eventhough i made sure to add the family relation to the new user table.

You should be able to access the relation via `@request.auth.family`, although there is no autocomplete since now we support multiple auth collections and the fields could be different (in future I'll consider loading the auth collection fields if there is only one).

**Reply by charbs-io** [reply]
Oh man... That was it 😅 I made a bad assumption that if it was not in autocomplete, it was not supported. That fixed it. Also, #911 seems to have affected me somewhat - thanks for pointing that out. 

My issue is now resolved, appreciate the prompt feedback, and really, really appreciate your work on this project. Can't wait to see where it goes.
